pallets/parachain-staking/src/lib.rs

```diff
@@ -1568,11 +1422,10 @@ pub mod pallet {
 			// should never fail but let's be safe
 			ensure!(insert_delegator, Error::<T>::DelegatorExists);
 
-			// can only throw if MaxCollatorsPerDelegator is set to 0 which should never
-			// occur in practice, even if the delegator rewards are set to 0
-			let delegator_state = Delegator::try_new(collator.clone(), amount)
-				.map_err(|_| Error::<T>::MaxCollatorsPerDelegatorExceeded)?;
-
+			let delegator_state = Delegator {
```

Comment:
We had to `try` before, because a `DelegatorState` included an ordered set of delegations. Now it's just `{ amount: BalanceOf<T>, owner: Option<AccountId> }`.